libs/editor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -1488,14 +1488,14 @@ public void onMediaUploadProgress(final String localMediaId, final float progres
 
     @Override
     public void onMediaUploadFailed(final String localMediaId, String errorType) {
-        switch (errorType) {
-            case "CONNECTION_ERROR":
-                getGutenbergContainerFragment().mediaFileUploadPaused(Integer.valueOf(localMediaId));
-                break;
-            default:
-                getGutenbergContainerFragment().mediaFileUploadFailed(Integer.valueOf(localMediaId));
-                break;
-        }
+        getGutenbergContainerFragment().mediaFileUploadFailed(Integer.valueOf(localMediaId));
+        mFailedMediaIds.add(localMediaId);
+        mUploadingMediaProgressMax.remove(localMediaId);
+    }
+
+    @Override
+    public void onMediaUploadPaused(final String localMediaId) {
+        getGutenbergContainerFragment().mediaFileUploadPaused(Integer.valueOf(localMediaId));
         mFailedMediaIds.add(localMediaId);
```

Comment:
I don't really wish to add these localMediaIds to `mFailedMediaIds`, but it is necessary to automatically restart media uploads:

https://github.com/wordpress-mobile/WordPress-Android/blob/234db297728a506ee8e9c318d42c9cd9e7e941b3/libs/editor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java#L1591-L1596

Semantically, this should probably be called `mFailedOrPausedMediaIds` (including related artifacts 
like `hasFailedMediaUploads` -> `hasFailedOrPausedMediaUploads`). The concept of failed uploads could be rethought of entirely with the introduction of `MEDIA_UPLOAD_STATE_PAUSED`. That feels well outside the scope of this change, as it has quite a large surface area. If you note a more elegant way to restart media uploads without adding these to `mFailedMediaIds`, I am open to it.

Reply:
I agree a refactor is likely beneficial but unnecessary at this time. 

That said, I do not believe overloading `mFailedMediaIds` to include "paused" media is all that cryptic or misleading, particularly if we only consider media to be "paused" if it also contains an error, as I noted in a separate comment.